holoviews/core/data/pandas.py

```diff
@@ -72,7 +74,7 @@ def init(cls, eltype, data, kdims, vdims):
 
             if kdims and not (len(kdims) == len(index_names) and {dimension_name(kd) for kd in kdims} == set(index_names)):
                 kdim = dimension_name(kdims[0])
-                if eltype._auto_indexable_1d and ncols == 1 and kdim not in data.columns:
+                if eltype._auto_indexable_1d and ncols == 1 and kdim not in chain(data.columns, cls.indexes(data)):
```

Comment:
```suggestion
                if eltype._auto_indexable_1d and ncols == 1 and kdim not in [*data.columns, *cls.indexes(data)]:
```

Do you think this will work?

Reply:
I think so. Which approach do you prefer?

Reply:
I think `chain` is a big tool for the job and adds an import. But I will let you choose, as I don't have that strong of an opinion.